micro-rdk/src/esp32/entry.rs

```diff
@@ -121,16 +121,15 @@ pub async fn serve_web_inner<S>(
                     robot
                 }
                 Err(err) => {
+                    log::error!("could not build robot: {:?}", err);
                     if let Some(datetime) = cfg_received_datetime {
                         let logs = vec![config_log_entry(datetime, Some(err))];
                         app_client
                             .push_logs(logs)
                             .await
                             .expect("could not push logs to app");
                     }
-                    // TODO(RSDK-8172) shouldn't panic here, when we support offline mode and
-                    // reloading configuration this should be removed
-                    panic!("couldn't build robot");
+                    LocalRobot::new()
```

Comment:
- I think the logged error should state that we couldn't build the robot from the *config* served by app, and that we are defaulting to creating an empty robot.
- Is there a way we can avoid writing the log push block twice?
- We probably shouldn't `expect` on the log push either since that will panic if it fails.

All these comments for the native entry as well.